controllers/disruption_controller.go

```diff
@@ -600,32 +600,27 @@ func (r *DisruptionReconciler) selectTargets(instance *chaosv1beta1.Disruption)
 		}
 	}
 
-	// prune ignored targets from all targets
-	for _, target := range matchingTargets {
-		if !contains(instance.Status.IgnoredTargets, target) {
-			eligibleTargets = append(eligibleTargets, target)
-		}
-	}
-
 	// return an error if the selector returned no targets
-	if len(eligibleTargets) == 0 {
+	if len(matchingTargets) == 0 {
 		r.log.Info("the given label selector did not return any targets, skipping")
 		r.Recorder.Event(instance, "Warning", "NoTarget", "The given label selector did not return any targets. Please ensure that both the selector and the count are correct (should be either a percentage or an integer greater than 0).")
 
 		return nil
 	}
 
+	// prune ignored targets from all targets
+	for _, target := range matchingTargets {
+		if !contains(instance.Status.IgnoredTargets, target) {
+			eligibleTargets = append(eligibleTargets, target)
+		}
+	}
+
 	// instance.Spec.Count is a string that either represents a percentage or a value, we do the translation here
-	targetsCount, err := getScaledValueFromIntOrPercent(instance.Spec.Count, len(eligibleTargets), true)
+	targetsCount, err := getScaledValueFromIntOrPercent(instance.Spec.Count, len(matchingTargets), true)
 	if err != nil {
 		targetsCount = instance.Spec.Count.IntValue()
 	}
 
-	// computed count should not be 0 unless the given count was not expected
```

Comment:
`targetsCount` can no longer be 0 here, so we don;t need this check. This was returning an error incorrectly in the case where `ignoredTargets` == `matchingTargets` anyway.